drools/src/main/java/com/datatorrent/drools/operator/StatelessDroolsOperator.java

```diff
@@ -140,4 +164,124 @@ public StatelessKieSession getKieSession()
   {
     return kieSession;
   }
+
+  private class RulesFiredListener implements AgendaEventListener
+  {
+
+    RulesFiredListener()
+    {
+
+    }
+
+    @Override
+    public void matchCreated(MatchCreatedEvent matchCreatedEvent)
+    {
+
+    }
+
+    @Override
+    public void matchCancelled(MatchCancelledEvent matchCancelledEvent)
+    {
+
+    }
+
+    @Override
+    public void beforeMatchFired(BeforeMatchFiredEvent event)
+    {
+
+    }
+
+    /**
+     * This function will be called automatically after firing a Rule.
+     * Modifies the map factsAndFiredRules and ruleCount.
+     * @param afterMatchFiredEvent the event which triggered this call
+     */
+    @Override
+    public void afterMatchFired(AfterMatchFiredEvent afterMatchFiredEvent)
+    {
+      Rule matchedRule = afterMatchFiredEvent.getMatch().getRule();
+      for (Object matchedObject : afterMatchFiredEvent.getMatch().getObjects()) {
+        List<Rule> currentRulesList = factsAndFiredRules.get(matchedObject);
+        currentRulesList.add(matchedRule);
+      }
+      Integer matchedRuleCount = ruleCount.get(matchedRule);
+      if (matchedRuleCount != null) {
+        matchedRuleCount++;
+      } else {
+        matchedRuleCount = 1;
+      }
+      ruleCount.put(matchedRule, matchedRuleCount);
+    }
+
+    @Override
+    public void agendaGroupPopped(AgendaGroupPoppedEvent agendaGroupPoppedEvent)
+    {
+
+    }
+
+    @Override
+    public void agendaGroupPushed(AgendaGroupPushedEvent agendaGroupPushedEvent)
+    {
+
+    }
+
+    @Override
+    public void beforeRuleFlowGroupActivated(RuleFlowGroupActivatedEvent ruleFlowGroupActivatedEvent)
+    {
+
+    }
+
+    @Override
+    public void afterRuleFlowGroupActivated(RuleFlowGroupActivatedEvent ruleFlowGroupActivatedEvent)
+    {
+
+    }
+
+    @Override
+    public void beforeRuleFlowGroupDeactivated(RuleFlowGroupDeactivatedEvent ruleFlowGroupDeactivatedEvent)
+    {
+
+    }
+
+    @Override
+    public void afterRuleFlowGroupDeactivated(RuleFlowGroupDeactivatedEvent ruleFlowGroupDeactivatedEvent)
+    {
+
+    }
+  }
+
+  /**
+   * Listener to track when new facts are added in the Session.
+   * Required because even rules can add new facts to the session.
+   */
+  private class FactsListener implements RuleRuntimeEventListener
```

Comment:
Same again can we extend DefaultRuleRuntimeEventListener and also remove the default constructor?